Assets/_Challenges/Scripts/AI Behaviours/AiBehaviour.cs

```diff
@@ -7,13 +7,36 @@ namespace MessyCoderCommunity.AI
 {
     public class AiBehaviour : ScriptableObject
     {
+        [SerializeField, Tooltip("The tick frequency in seconds. Set to zero for every frame.")]
+        float tickFrequency = 0f;
+
+        private float nextTickTime = 0;
+
         public virtual Type AgentType {
             get { return typeof(GameObject); }
         }
 
         public virtual void Initialize(GameObject agent, Chalkboard chalkboard) { }
 
-        public virtual void Tick(Chalkboard chalkboard) { }
+        public virtual void Tick(Chalkboard chalkboard)
```

Comment:
I generally like to use protected OnAction for methods that are overridable and Action() for the method that is callable. This allows user to forget base.Action() without consequence. 

I think what I'd like to see here would be more like:
internal void Tick(Chalkboard c) {
    if(IsTimeToUpdate) OnTick(Chalkboard chalkboard)
}

protected virtual void OnTick(Chalkboard) {}